api/src/main/java/com/github/retrooper/packetevents/util/PEVersion.java

```diff
@@ -18,142 +18,212 @@
+            throw new IllegalArgumentException("Version string must be in the format 'major.minor.patch[-SNAPSHOT]'");
         }
+
+        this.major = Integer.parseInt(parts[0]);
+        this.minor = Integer.parseInt(parts[1]);
+        this.patch = Integer.parseInt(parts[2]);
     }
 
     /**
-     * Compare to another PEVersion.
-     * If we are newer than the compared version,
-     * this method will return 1.
-     * If we are older than the compared version,
-     * this method will return -1.
-     * If we are equal to the compared version,
-     * this method will return 0.
-     * Similar to {@link Integer#compareTo(Integer)}.
-     *
-     * @param version Compared version
-     * @return Comparing to another Version.
-     */
-    public int compareTo(PEVersion version) {
-        int localLength = versionIntArray.length;
-        int oppositeLength = version.versionIntArray.length;
-        int length = Math.max(localLength, oppositeLength);
-        for (int i = 0; i < length; i++) {
-            int localInteger = i < localLength ? versionIntArray[i] : 0;
-            int oppositeInteger = i < oppositeLength ? version.versionIntArray[i] : 0;
-            if (localInteger > oppositeInteger) {
-                return 1;
-            } else if (localInteger < oppositeInteger) {
-                return -1;
-            }
-        }
-        return 0;
+     * Creates a {@link PEVersion} instance from the package implementation version.
+     *
+     * @return a {@link PEVersion} instance.
+     */
+    public static PEVersion createFromPackageVersion() {
+        String version = Optional.ofNullable(PacketEvents.class.getPackage().getImplementationVersion()).orElse("0.0.0");
+        return new PEVersion(version);
+    }
+
+    /**
+     * Gets the major version number.
+     *
+     * @return the major version number.
+     */
+    public int major() {
+        return major;
     }
 
     /**
-     * Does the {@link #compareTo(PEVersion)} return 1?
+     * Gets the minor version number.
      *
-     * @param version Compared version.
-     * @return Is this newer than the compared version.
+     * @return the minor version number.
      */
-    public boolean isNewerThan(PEVersion version) {
-        return compareTo(version) == 1;
+    public int minor() {
+        return minor;
     }
 
     /**
-     * Does the {@link #compareTo(PEVersion)} return -1?
+     * Gets the patch version number.
      *
-     * @param version Compared version.
-     * @return Is this older than the compared version.
+     * @return the patch version number.
      */
-    public boolean isOlderThan(PEVersion version) {
-        return compareTo(version) == -1;
+    public int patch() {
+        return patch;
     }
 
     /**
-     * Represented as an array.
+     * Checks if the version is a snapshot.
      *
-     * @return Array version.
+     * @return true if snapshot, false otherwise.
      */
-    public int[] asArray() {
```

Comment:
this method has been removed, is this an intended breaking change?

Reply:
Yes, I wasn't sure how to properly implement it without just skipping the snapshot part. Do you want me to add it, and mark it as deprecated? This would mean snapshot won't be accounted for in the return value of that method.

Reply:
:+1: snapshot wasn't present there previously either, an `@Deprecated` is definetely needed

Reply:
I'll add this tomorrow morning, and test this while shading PE as well!

Reply:
You've called it "toArray", it should be named "asArray" 😄

Reply:
Fixed 😅